### DISCUSSION TITLE
Close subscription from server

### DISCUSSION THREAD

**Discussion by gunhaxxor** [trpc/trpc#3779] [Q&A]
Is it possible to close subscriptions from the server side? If so, how?

**Comment by KATT** [comment]
Yes, you can emit an event to ask the clients to reconnect: https://github.com/trpc/trpc/blob/949f42b900291a3985c178d8043ec1ffccc9e235/examples/next-prisma-websockets-starter/src/server/prodServer.ts#L32-L35

You can also reach into the underlying `ws`-lib and force-close the clients

**Comment by gunhaxxor** [comment]
I believe you misunderstood the question. The question is how to close/complete an active subscription. Perhaps Iv'e misunderstood the terminology?
Either way, here's a scenario that hopefully clarifies:
Server has a list of users in a room. A client joins the room and at the same time starts subscribing to events related to the room.
After a while the server kicks the user out of the room (not from the server and not closing the ws connection, only the room). The server should be able to "complete" the subscription and thus stopping the client from receiving subscription events related to the room.

Since I asked this question I've come to the understanding that this perhaps could be achieved by calling the "complete" function on the subscriber created by the observable? I haven't tried this though. Another user recently asked a similar question and stated it didn't work for them. Is calling complete from server supported?
https://github.com/trpc/trpc/discussions/4060

**Reply by KATT** [reply]
Yeah, `complete` should do it I think.


I haven't used our subscriptions for over a year, if it doesn't work, please open a feature request
